File: backend/internal/routes/analytics.go
package routes

import (
	"github.com/ZerkerEOD/krakenhashes/backend/internal/db"
	"github.com/ZerkerEOD/krakenhashes/backend/internal/handlers/analytics"
	"github.com/ZerkerEOD/krakenhashes/backend/internal/services"
	"github.com/ZerkerEOD/krakenhashes/backend/pkg/debug"
	"github.com/gorilla/mux"
)

// SetupAnalyticsRoutes configures routes for password analytics
func SetupAnalyticsRoutes(router *mux.Router, database *db.DB, queueService *services.AnalyticsQueueService) {
	debug.Info("Setting up analytics routes...")

	// Create handler
	handler := analytics.NewHandler(database, queueService)

	// Analytics routes (all require authentication via JWT middleware)
	router.HandleFunc("/analytics/clients", handler.GetClients).Methods("GET", "OPTIONS")
	router.HandleFunc("/analytics/reports", handler.CreateReport).Methods("POST", "OPTIONS")
	router.HandleFunc("/analytics/reports/{id}", handler.GetReport).Methods("GET", "OPTIONS")
	router.HandleFunc("/analytics/reports/client/{clientId}", handler.GetClientReports).Methods("GET", "OPTIONS")
	router.HandleFunc("/analytics/reports/{id}", handler.DeleteReport).Methods("DELETE", "OPTIONS")
	router.HandleFunc("/analytics/reports/{id}/retry", handler.RetryReport).Methods("POST", "OPTIONS")
	router.HandleFunc("/analytics/queue-status", handler.GetQueueStatus).Methods("GET", "OPTIONS")

	debug.Info("Analytics routes configured successfully")
}

File: frontend/src/services/analytics.ts
import { api } from './api';
import {
  AnalyticsReport,
  CreateAnalyticsReportRequest,
  QueueStatus,
} from '../types/analytics';

export const analyticsService = {
  // Create a new analytics report
  createReport: async (data: CreateAnalyticsReportRequest): Promise<AnalyticsReport> => {
    const response = await api.post('/api/analytics/reports', data);
    return response.data;
  },

  // Get a specific report by ID
  getReport: async (id: string): Promise<{ status: string; message?: string; report: AnalyticsReport }> => {
    const response = await api.get(`/api/analytics/reports/${id}`);
    return response.data;
  },

  // Get all reports for a specific client
  getClientReports: async (clientId: string): Promise<AnalyticsReport[]> => {
    const response = await api.get(`/api/analytics/reports/client/${clientId}`);
    return response.data;
  },

  // Delete a report
  deleteReport: async (id: string): Promise<void> => {
    await api.delete(`/api/analytics/reports/${id}`);
  },

  // Retry a failed report
  retryReport: async (id: string): Promise<AnalyticsReport> => {
    const response = await api.post(`/api/analytics/reports/${id}/retry`);
    return response.data;
  },

  // Get queue status
  getQueueStatus: async (): Promise<QueueStatus> => {
    const response = await api.get('/api/analytics/queue-status');
    return response.data;
  },
};

export default analyticsService;
